plugins/kn-image/cmd/deploy.go

```diff
@@ -0,0 +1,276 @@
+/*
+Copyright © 2019 NAME HERE <EMAIL ADDRESS>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"fmt"
+	"github.com/knative/client-contrib/plugins/kn-image/tekton"
+	servingclientset_v1alpha1 "github.com/knative/client/pkg/serving/v1alpha1"
+	serving_v1alpha1_api "github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	serving_v1beta1_api "github.com/knative/serving/pkg/apis/serving/v1beta1"
+	serviceclientset "github.com/knative/serving/pkg/client/clientset/versioned"
+	"github.com/spf13/cobra"
+	tektoncdclientset "github.com/tektoncd/pipeline/pkg/client/clientset/versioned"
+	corev1 "k8s.io/api/core/v1"
+	api_errors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/oidc" // from https://github.com/kubernetes/client-go/issues/345
+	"k8s.io/client-go/tools/clientcmd"
+	"os"
+	"time"
+)
+
+// deployCmd represents the deploy command
+var deployCmd = &cobra.Command{
+	Use:   "deploy",
+	Short: "Build the container image from source code or deploy Knative service",
+	Example: `
+  # Deploy from git repository to Knative service
+  # ( related: https://github.com/knative/client-contrib/blob/master/plugins/kn-image/doc/deploy-git-resource.md )
+  kn-imaage deploy cnbtest --builder buildpacks-v3 --git-url https://github.com/zhangtbj/cf-sample-app-nodejs --git-revision master --saved-image us.icr.io/test/cnbtest:v1 --serviceaccount default --force`,
+	Run: func(cmd *cobra.Command, args []string) {
```

Comment:
Same as before this is way too long of a function. Please split into smaller private functions which will make clear what the steps are.

Reply:
Did some refine.